### PR TITLE
Enhance customisability of modal

### PR DESCRIPTION
**Pull request by stefan-squareweave**
Adds template and JS hooks to facilitate the customisation of the modal popup.

### CODE REVIEW

**Comment by seb-b** [comment]
This is good, well documented which makes it easier. Hopefully I'll get a chance to have a proper look at this soon

File: wagtailmodelchooser/static/wagtailmodelchooser/js/model_chooser.js [hunk 1]
```diff
@@ -32,59 +82,86 @@ function setupModal(modal, jsonData) {
...
      */
     var searchForm = modal.body.find('form.search-bar');
-    var searchField = searchForm.find('#id_q');
     var searchUrl = searchForm.attr('action');
     var searchResults = modal.body.find('#search-results');
+    // Builds a dictionary of the values of fields in searchForm
+    function buildSearchDict() {
+        var rtn = {};
+        searchForm.find("input, select, textarea").each(function() {
+            rtn[$(this).attr("name")] = $(this).val();
```
Comment (seb-b) [inline wagtailmodelchooser/static/wagtailmodelchooser/js/model_chooser.js]: I don't think this covers boolean form controls like checkboxes and radios, value will be the value not which on is checked.

You could possibly use the `FormData` class to easily build up a dict

https://developer.mozilla.org/en-US/docs/Web/API/FormData

File: wagtailmodelchooser/templates/wagtailmodelchooser/modal.html [hunk 2]
```diff
@@ -11,13 +11,19 @@
 <div class="nice-padding">
 	{# Need to keep the form in the HTML, even if the snippet is not searchable #}
 	{# This is to allow pagination links to be generated from the form action URL #}
-	<form class="search-bar" action="{{ chooser_url }}" method="GET">
+	{% block search_form %}
+	<form class="search-bar" action="{{ chooser_url }}" method="GET" style="margin-bottom:20px">
```
Comment (seb-b) [inline wagtailmodelchooser/templates/wagtailmodelchooser/modal.html]: Inline styles aren't ideal, I think from memory this used to pretty similar to the current wagtail modal, is there any classes we can leverage from there? I'm not too fussed if not

File: wagtailmodelchooser/static/wagtailmodelchooser/js/model_chooser.js (hunk 1 above)
Comment (stefan-squareweave) [reply wagtailmodelchooser/static/wagtailmodelchooser/js/model_chooser.js]: Thanks for the pickup. Had a go with FormData, but the API is weird. Added a case for radios/checkboxes using jQuery.

File: wagtailmodelchooser/templates/wagtailmodelchooser/modal.html (hunk 2 above)
Comment (stefan-squareweave) [reply wagtailmodelchooser/templates/wagtailmodelchooser/modal.html]: Copied some HTML/classes from the image selector modal. No more inline styles!

**Comment by seb-b** [comment]
Thanks for this, released in 2.3.0

**Pull request merged**